packages/commons-server/src/libs/server/server.ts

```diff
@@ -904,4 +923,92 @@ export class MockoonServer extends (EventEmitter as new () => TypedEmitter<Serve
 
     return tlsOptions;
   }
+
+  private generateDatabuckets(environment: Environment) {
+    if (environment.data.length > 0) {
+      environment.data.forEach((databucket) => {
+        let newProcessedDatabucket: ProcessedDatabucket;
+
+        if (
+          databucket.value.match(
+            new RegExp(`\{\{[\#\w\s\(]*(${listOfRequestHelperTypes.join('|')})`)
+          )
+        ) {
+          // a request helper was found
+          newProcessedDatabucket = {
+            name: databucket.name,
+            value: databucket.value,
+            parsed: false
+          };
+          newProcessedDatabucket.value = databucket.value;
+          newProcessedDatabucket.parsed = false;
+        } else {
+          const templateParsedContent = TemplateParser(
+            false,
+            databucket.value,
+            environment,
+            this.processedDatabuckets
+          );
+
+          try {
+            const JSONParsedContent = JSON.parse(templateParsedContent);
+            newProcessedDatabucket = {
+              name: databucket.name,
+              value: JSONParsedContent,
+              parsed: true
+            };
+          } catch (e) {
+            newProcessedDatabucket = {
+              name: databucket.name,
+              value: templateParsedContent,
+              parsed: true
+            };
+          }
+        }
+        this.processedDatabuckets.push(newProcessedDatabucket);
+      });
+    }
+  }
+
+  private generateCalledDatabucket(
+    route: Route,
+    environment: Environment,
+    request: Request
+  ) {
+    route.responses.forEach((response) => {
+      const results = response.body?.matchAll(
+        new RegExp('{{2,3}data [\'|"]{1}([^(\'|")]*)', 'g')
```

Comment:
I think we are not accounting for #, spaces and parenthesis that could follow the `{{` in this regex (unlike the other one).